runtime/lua/vim/lsp/completion.lua

```diff
@@ -630,6 +630,105 @@ local function get_augroup(bufnr)
   return string.format('nvim.lsp.completion_%d', bufnr)
 end
 
+---@param winid integer
+---@param bufnr integer
+---@param ft? string
+local function update_popup_window(winid, bufnr, ft)
+  if winid and api.nvim_win_is_valid(winid) and bufnr and api.nvim_buf_is_valid(bufnr) then
+    vim.wo[winid].conceallevel = 2
+    vim.bo[bufnr].filetype = ft or ''
```

Comment:
Do we need to set filetype for that buffer (and then executing autocommands there)? Is not `vim.treesitter.start(bufnr, vim.treesitter.language.get_lang(filetype))` enough?

Reply:
because it returns `string | markdown | plaintext`.. so can only use ts to cover all cases? At least it can fall back to the traditional syntax if there have when set ft

Reply:
Lots of plugins use `au FileType *` anti-pattern, so it would be good to reduce frequency of sending such event. What about
```suggestion
    if vim.bo[bufnr].filetype ~= ft then
      vim.bo[bufnr].filetype = ft or ''
    end
```
?

Reply:
make sense